src/irr/asset/CImageWriterPNG.cpp

```diff
@@ -113,20 +112,20 @@ bool CImageWriterPNG::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 	{
 		case asset::EF_R8G8B8_SRGB:
 			png_set_IHDR(png_ptr, info_ptr,
-				image->getSize().X, image->getSize().Y,
+				trueExtent.X, trueExtent.Y,
 				8, PNG_COLOR_TYPE_RGB, PNG_INTERLACE_NONE,
 				PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
 			break;
 		case asset::EF_R8G8B8A8_SRGB:
 			png_set_IHDR(png_ptr, info_ptr,
-				image->getSize().X, image->getSize().Y,
+				trueExtent.X, trueExtent.Y,
 				8, PNG_COLOR_TYPE_RGB_ALPHA, PNG_INTERLACE_NONE,
 				PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
 		break;
 		case asset::EF_R8_SRGB:
 		case asset::EF_R8_UNORM:
```

Comment:
now you definitely want to only handle `RF_R8_SRGB` as pass-through

Reply:
not `UNORM`

Reply:
I don't know in that why it was as that previously